server/context.go

```diff
@@ -227,9 +230,8 @@ func (s *SessionManager) NewContextWithQuery(conn *mysql.Conn, query string) (*s
 func (s *SessionManager) KillConnection(connID uint32) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	if entry, ok := s.sessions[connID]; ok {
-		delete(s.sessions, connID)
-		entry.conn.Close()
+	if conn, ok := s.connections[connID]; ok {
+		conn.Close()
```

Comment:
It wasn't immediately clear to me why we no longer delete the connection from the sessions when we kill it. I'm guessing that the semantics for KillConnection should be that it destroys the connection, but does NOT remove it from the set of connections it knows about and only `CloseConn` removes the state. If that's true, might be worth mentioning in `KillConnections` docstring that it only closes the connection and callers must call `CloseConn` to remove the state... although... `CloseConn` doesn't actually close the connection, it just removes the state for it? Seems like `CloseConn` is only called from the handler when the connection has already been closed... so maybe `RemoveConnState` would be a more accurate name?

Reply:
btw... I realize the naming wasn't part of your change at all, so totally fair to defer that suggestion for a different PR of course 😄

Reply:
Yeah, the semantics of these is asynchronous termination of the running query or the connection, and it felt a lot more correct to say there's only one place where we remove the connection/session from the manager and that's when the server actually marks it disconnected. I'll take a pass at cleaning up naming and documentation.